shpc/main/settings.py

```diff
@@ -178,8 +183,8 @@ def set(self, key, value):
             logger.exit("You cannot use 'set' for a list. Use add/remove instead.")
 
         # This is a reference to a dictionary (object) setting
-        if ":" in key:
-            key, subkey = key.split(":")
+        if ":" in value:
+            subkey, value = value.split(":")
```

Comment:
And I could have also put `.split(":", 1)` here, the assumption being that we only have one level of nesting, but I left it like this because I'd rather it error out and give the user a message (and then deal with it) then to silently pass another layer of `:` into a config.

Reply:
Yep, makes sense!

For the `get` function , lines 118-122, does my edit seems sensible to you?

Reply:
everything except for the fact that we want to check the value and not the key - the key (by way of the split on the first :) won't have the second ":" - but I think you probably wrote it given the reverse in config.py, so my changes here should resolve both those.

Reply:
oh no wait I totally missed that - that is wrong too.

Reply:
Oh wait I think I understand - you want to do like:

```python
settings.get('container_features:gpu')
# amd
```
vs.

```python
settings.get('container_features')['gpu']
```
I was thinking the second is more straight forward in terms of parsing python, but if you are interested just in the value I can see the first being okay. It's different than the second parsing (set) where the second ":" is in the value but if it works I have no issue with it. Let me know your thoughts!

Reply:
I think we need whichever implementation makes the following CLI command to work: `shpc config get container_features:gpu`

Reply:
okay then I think what you have is perfecto - checking for the : in the key, and then splitting based on that. I'll add a test for it, and it that passes (and all is okay with you) merge and release.